### PR TITLE
Update for ATMS for End-to-End Test

### PR DESCRIPTION
**Pull request by emilyhcliu**
List of items to be added in this PR:
- BUFR Converter: Add YAML for converting ATMS BUFR to IODA
   - The ObsValue is antenna temperature
   - Input BUFR is normal feel 

- Testing YAML: re-evaluation results due to update CRTM from v2.3.0 to v2.4.1-jedi.1
   - atms_npp_noqc.yaml --- 100% replication
   - atms_npp.yaml  --- 100 % replication
   - atms_n20_noqc.yaml --- 100 % replication
   - atms_n20.yaml ---  ~ 1100% replication: two channel 7 observations difference
  
- Config YAML: 
   * add two read routine QC: data thinning and removal of data from scan edge
   - atms_npp.yaml
   - atms_n20.yaml   

- Paring PR in JCSDA UFO repository
   - This PR also has a [paring PR in JCSDA UFO repository](https://github.com/JCSDA-internal/ufo/pull/3094).
   - It is OK to merge this one into GDASApp before the paring UFO PR
   - The related test data (obs and geoval files from GSI) for UFO Evaluation have been updated (due to CRTM-2.4.1 update) on ORION and HERA.

Notes:
- End-to-end testing completed without problem. 
- However, the filter results show one of the filters, 88-165 GHz scattering index check, tossed more data compared to GSI. 
- The excess screening from the 88-165 GHz scattering index check is under investigation and will be reported separately.
- Please check comments below for validation results

### CODE REVIEW

**Comment by emilyhcliu** [comment]
### UFO evaluation for CRTM-2.4.1 update

**ATMS NPP**
**Results without QC**
**HofX Difference  (No QC) Comparison (JEDI vs. GSI) for all channels**
HofX_Difference_atms_npp_diag_2021080100_atms_npp

**Initial Obs Error Difference (JEDI vs. GSI) for all channels (No QC)**
ObsErr_Difference_atms_npp_diag_2021080100_atms_npp

**Results with QC**
**HofX Difference with QC (JEDI vs. GSI) for all channels**
HofX_Difference_atms_npp_diag_2021080100_atms_npp

**Obs Erro Difference with QC**
ObsErr_Difference_atms_npp_diag_2021080100_atms_npp

**QC Difference (JEDI vs. GSI) for all channels**
QCCount_Diff_atms_npp_diag_2021080100_atms_npp

Note: 100% replication of GSI QC

**Comment by emilyhcliu** [comment]
### UFO evaluation for CRTM-2.4.1 update

**ATMS N20**
**Results without QC**
**HofX Difference (No QC) Comparison (JEDI vs. GSI) for all channels**
HofX_Difference_atms_n20_diag_2021080100_atms_n20

**Initial Obs Error Difference (JEDI vs. GSI) for all channels (No QC)**
ObsErr_Difference_atms_n20_diag_2021080100_atms_n20

This is Initial Obs Error Assignment for  ATMS N20

ufo_atms_n20_obserr0_bc_channel_2

**Results with QC**
**HofX Difference with QC Comparison (JEDI vs. GSI) for all channels**
HofX_Difference_atms_n20_diag_2021080100_atms_n20

**Obs Error Difference with QC Comparison (JEDI vs. GSI) for all channels**
ObsErr_Difference_atms_n20_diag_2021080100_atms_n20

**QC Difference (JEDI vs. GSI) for all channels**
QCCount_Diff_atms_n20_diag_2021080100_atms_n20

Notes: **Two observations** difference from channel 7

**Review by ADCollard — APPROVED** [review]

File: parm/atm/obs/lists/gdas_prototype_3d.yaml [hunk 1]
```diff
@@ -2,12 +2,16 @@ observers:
 ##- !INC ${OBS_YAML_DIR}/amsua_n19.yaml
 ##- !INC ${OBS_YAML_DIR}/sondes_prepbufr.yaml
 ##- !INC ${OBS_YAML_DIR}/atms_n20.yaml
+##- !INC ${OBS_YAML_DIR}/atms_npp.yaml
```
Comment (RussTreadon-NOAA) [inline parm/atm/obs/lists/gdas_prototype_3d.yaml:5]: How do we want gdas-validation developers to update `gdas_prototype_3d.yaml` as they merge their gdas-validation work into `feature/gdas-validation`?   

1. Add new observation types to the file but keep them commented out (inactive) ... as is done this PR #757, or
2. Add new observations types to the file in active mode (don't comment them out) ... as is done in PR #753.
3. don't change `gdas_prototype_3d.yaml`

**Comment by emilyhcliu** [comment]
### End-to-End   (using ATMS NPP channel 2 as an example)

**All Data**
ufo_atms_npp_omf_bc_channel_2


**Comparison QC between UFO and GSI**
**The UFO (end-to-end) filters out more data.  The scattering index check (88-165 difference) tosses more data.**  
(See the dark green color from the qc flag comparison below)

**UFO after QC**
ufo_atms_npp_omf_qc_channel_2

**GSI after QC**
gsi_atms_npp_omf_gsi_qc_channel_2

**Comparison QC Flags between UFO and GSI**
**UFO QCFlags**
ufo_atms_npp_qcflag_channel_2

**GSI QC Flags**
gsi_atms_npp_qcflag_channel_2

**Review by RussTreadon-NOAA — APPROVED** [review]
Ran 2021080100 gdas cycle from prepatmiodaobs through atmanlfinal.   All jobs ran to completion.  atms_n20 and atms_npp data processed.   Increments look reasonable.   
```
 0: Increment print | number of fields = 8 | cube sphere face size: C768
  0: eastward_wind                                | Min:-4.611234e-09 Max:+4.406635e-09 RMS:+6.948029e-12
  0: northward_wind                               | Min:-5.030879e-09 Max:+5.978815e-09 RMS:+7.534733e-12
  0: air_temperature                              | Min:-3.008589e-07 Max:+2.911516e-07 RMS:+1.419730e-10
  0: surface_pressure                             | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
  0: specific_humidity                            | Min:-3.916608e-04 Max:+2.488543e-03 RMS:+7.406468e-07
  0: cloud_liquid_ice                             | Min:+0.000000e+00 Max:+1.618770e-20 RMS:+1.293217e-23
  0: cloud_liquid_water                           | Min:+0.000000e+00 Max:+1.474788e-19 RMS:+2.167418e-22
  0: ozone_mass_mixing_ratio                      | Min:+0.000000e+00 Max:+0.000000e+00 RMS:+0.000000e+00
```
Final diagnostic tarball contains atms diagnostic files.  
```
Orion-login-2:/work2/noaa/da/rtreadon/gdas-validation/comrot/gdas_eval_satwind_JEDI/gdas.20210801/00/analysis/atmos$ tar -tvf gdas.t00z.atmstat
-rw-r--r-- rtreadon/stmp 6888137 2023-11-20 20:17 diag_satwind_ahi_h8_2021080100.nc4.gz
-rw-r--r-- rtreadon/stmp 5797597 2023-11-20 20:17 diag_ascatw_ascat_metop-b_2021080100.nc.gz
-rw-r--r-- rtreadon/stmp 36257317 2023-11-20 20:17 diag_gnssro_2021080100.nc4.gz
-rw-r--r-- rtreadon/stmp 716249805 2023-11-20 20:19 diag_atms_npp_2021080100.nc.gz
-rw-r--r-- rtreadon/stmp  52721306 2023-11-20 20:19 diag_satwind_abi_goes-16_2021080100.nc.gz
-rw-r--r-- rtreadon/stmp  95641266 2023-11-20 20:20 diag_satwind_abi_goes-17_2021080100.nc.gz
-rw-r--r-- rtreadon/stmp   4746726 2023-11-20 20:20 diag_ascatw_ascat_metop-a_2021080100.nc.gz
-rw-r--r-- rtreadon/stmp 728609297 2023-11-20 20:22 diag_atms_n20_2021080100.nc.gz
```

Aside:  as we validate more observation types, the run time for gdasatmanlfinal increases due to gzipping observation diagnostic files.  With 8 files we are up to about 5 minutes wall time.   We will likely need to parallelize this at a later date.
```
^[[38;21m2023-11-20 20:17:19,097 - INFO     - atm_analysis: Compressing 8 diag files to /work2/noaa/da/rtreadon/gdas-validation/comrot/gdas_eval_satwind_JEDI/gdas.20210801/00//analysis/atmos/gdas.t00z.atmstat.gz^[[0m
^[[38;5;39m2023-11-20 20:17:19,097 - DEBUG    - atm_analysis: Gzipping 8 diag files^[[0m
^[[38;5;39m2023-11-20 20:22:07,498 - DEBUG    - atm_analysis: Creating tar file /work2/noaa/da/rtreadon/gdas-validation/comrot/gdas_eval_satwind_JEDI/gdas.20210801/00//analysis/atmos/gdas.t00z.atmstat with 8 gzipped diag files^[[0m
^[[38;21m2023-11-20 20:22:10,568 - INFO     - atm_analysis: Copying /work/noaa/stmp/rtreadon/RUNDIRS/gdas_eval_satwind_JEDI/gdasatmanl_00/gdas.t00z.atmvar.yaml to /work2/noaa/da/rtreadon/gdas-validation/comrot/gdas_eval_satwind_JEDI/gdas.20210801/00//analysis/atmos^[[0m
^[[38;5;39m2023-11-20 20:22:10,569 - DEBUG    - atm_analysis: Copying /work/noaa/stmp/rtreadon/RUNDIRS/gdas_eval_satwind_JEDI/gdasatmanl_00/gdas.t00z.atmvar.yaml to /work2/noaa/da/rtreadon/gdas-validation/comrot/gdas_eval_satwind_JEDI/gdas.20210801/00//analysis/atmos/gdas.t00z.atmvar.yaml^[[0m
```

Approve.

File: parm/atm/obs/lists/gdas_prototype_3d.yaml (hunk 1 above)
Comment (emilyhcliu) [reply parm/atm/obs/lists/gdas_prototype_3d.yaml:5]: Here is my two cents:
We can just leave satwind_g16 active as default and make the rest of the data type inactive.
The newly added data type will be inactive.

**Comment by emilyhcliu** [comment]
@xincjin-NOAA  and I will work on AMSU-A next...

**Comment by emilyhcliu** [comment]
The ATMS test data for UFO evaluation has been updated on both Orion and Hera.

**Comment by CoryMartin-NOAA** [comment]
The `gdas_prototype_3d.yaml` file has conflicts that must be resolved before this can be merged.

**Comment by emilyhcliu** [comment]
> The `gdas_prototype_3d.yaml` file has conflicts that must be resolved before this can be merged.

Resolved!

File: parm/atm/obs/lists/gdas_prototype_3d.yaml (hunk 1 above)
Comment (RussTreadon-NOAA) [reply parm/atm/obs/lists/gdas_prototype_3d.yaml:5]: Works for me.  As long as we all follow the same approach, we won't have any conflicts with `gdas_prototype_3d.yaml`.   We can turn on all or a subset of the validated data once the sprint concludes.

**Pull request merged**